### PR TITLE
Add `get_entry_points`, `EntryPoint` struct, and a test.

### PR DESCRIPTION
**Pull request by dwerner**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -154,6 +154,13 @@ pub struct PushConstantInfo {
     pub size: u32,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct EntryPoint {
+    pub execution_model: spirv::ExecutionModel,
+    pub id: u32,
```
Comment (MarijnS95) [inline src/lib.rs]: The type of this number is indeed an `IdRef` operand, but the name should refect what it's pointing to: the `OpFunction` that this `OpEntryPoint` is describing. Can you rename this to `function_id`?

File: src/lib.rs [hunk 2]
```diff
@@ -685,4 +692,23 @@ impl Reflection {
...
+    pub fn get_entry_points(&self) -> Result<Vec<EntryPoint>> {
+        let reflect = &self.0;
+        reflect
+            .entry_points
+            .iter()
+            .map(|entry_point| -> Result<EntryPoint> {
+                let execution_model = get_operand_at!(entry_point, Operand::ExecutionModel, 0)?;
+                let id = get_operand_at!(entry_point, Operand::IdRef, 1)?;
+                let name = get_ref_operand_at!(entry_point, Operand::LiteralString, 2)
+                    .map(Clone::clone)?;
```
Comment (MarijnS95) [inline src/lib.rs]: I think that's just `.cloned()`? But it should also be fine to return `EntryPoint<'_>` containing a borrowed `name: &'a str` with a lifetime on `&self`.

File: src/lib.rs [hunk 3]
```diff
@@ -685,4 +692,23 @@ impl Reflection {
         use rspirv::binary::Disassemble;
         self.0.disassemble()
     }
+
+    pub fn get_entry_points(&self) -> Result<Vec<EntryPoint>> {
+        let reflect = &self.0;
+        reflect
+            .entry_points
+            .iter()
+            .map(|entry_point| -> Result<EntryPoint> {
```
Comment (MarijnS95) [inline src/lib.rs]: Is the explicit result type needed here?

**Review by MarijnS95 — COMMENTED** [review]
Thanks for your contribution! Keep in mind though that `rspirv-reflect` is mainly intended to implement nontrivial reflection functionality on top of `rspirv`: we provide access to the nested `rspirv::dr::Module` for more straightforward queries.

File: src/lib.rs (hunk 2 above)
Comment (dwerner) [reply src/lib.rs]: Hah, thanks. DescriptorInfo is holding a string, so I followed the established way.

**Comment by dwerner** [comment]
> Thanks for your contribution! Keep in mind though that `rspirv-reflect` is mainly intended to implement nontrivial reflection functionality on top of `rspirv`: we provide access to the nested `rspirv::dr::Module` for more straightforward queries.

So are you saying is this a reflection that you'd prefer not to have?

**Comment by dwerner** [comment]
> https://docs.rs/rspirv/latest/rspirv/lift/index.html

Ah, so just incomplete documentation in rspirv on their separate dr and sr modules, and you compose the dr (data representation). I didn't dig as deep as I could have before just doing the conversion here.

It would have been more helpful to me if you made your blocker (and I assume it is that) comprehensible, rather than review and nitpick when you have no intention to merge a PR. You knew that a more sensible route exists.

**Comment by MarijnS95** [comment]
> You knew that a more sensible route exists.

Ah I wasn't aware of the SR layer on top of the DR layer already covering this, but yes it apparently already exists and returns a struct with the same fields: https://docs.rs/rspirv/latest/rspirv/sr/autogen_instructions/struct.EntryPoint.html

> It would have been more helpful to me if you made your blocker (and I assume it is that) comprehensible, rather than review and nitpick when you have no intention to merge a PR. You knew that a more sensible route exists.

I'd be fine to have this, but it's getting close to the "just get this directly from `rspirv`" territory (where e.g. descriptor set parsing and type-size gathering is nontrivial). We both just learned that the SR does this, though: I was _not_ aware of the existence when writing my review and was _not_ considering it a blocker.

**Comment by dwerner** [comment]
Gotcha. Sorry I misunderstood that.

**Pull request closed without merging**